clients/venice-client/src/main/java/com/linkedin/venice/fastclient/meta/HelixScatterGatherRoutingStrategy.java

```diff
@@ -15,8 +15,12 @@
 public class HelixScatterGatherRoutingStrategy implements ClientRoutingStrategy {
   private Map<String, Integer> helixGroupInfo;
   private List<Integer> groupIds;
+  private final InstanceHealthMonitor instanceHealthMonitor;
 
-  public HelixScatterGatherRoutingStrategy(Map<String, Integer> helixGroupInfo) {
+  public HelixScatterGatherRoutingStrategy(
+      InstanceHealthMonitor instanceHealthMonitor,
+      Map<String, Integer> helixGroupInfo) {
```

Comment:
Function definition looks odd. This implies that `helixGroupInfo` is setup during initing `HelixScatterGatherRoutingStrategy`, but that is the test's flow. non test flow actually inits  `HelixScatterGatherRoutingStrategy` with `new VeniceConcurrentHashMap<>()` and then uses `updateHelixGroupInfo` to replace that during the initial warmup. so I feel that it's cleaner like below
```
public HelixScatterGatherRoutingStrategy(InstanceHealthMonitor instanceHealthMonitor) {
....
}
```

Reply:
Yes, let me change it.